### PR TITLE
DO NOT MERGE - DDF-0000 Message

### PR DESCRIPTION
**Pull request by coyotesqrl**
#### What does this PR do?
Stuff.

#### Who is reviewing it? 

(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)

#### What are the relevant tickets?
[](https://codice.atlassian.net/browse/)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by coyotesqrl** [comment]
✏️ PR needs assigned committers.

File: catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/CreateRequestImpl.java [hunk 1]
```diff
@@ -66,7 +66,10 @@ public CreateRequestImpl(List<Metacard> metacards, Map<String, Serializable> pro
   public CreateRequestImpl(
       List<Metacard> metacards, Map<String, Serializable> properties, Set<String> destinations) {
     super(properties);
-    this.metacards = metacards;
+    this.metacards = new HashSet<>();
+    for (Metacard metacard : metacards) {
+      this.metacards.add(metacard);
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/CreateRequestImpl.java]: ❗️ Wrong! This is inefficient! Use the `addAll` method.

File: catalog/core/catalog-core-api/src/main/java/ddf/catalog/operation/CreateRequest.java [hunk 2]
```diff
@@ -26,9 +27,9 @@
...
 public interface CreateRequest extends Request {
   /**
-   * Gets a {@link List} of {@link Metacard} to be stored.
+   * Gets a {@link Set} of {@link Metacard} to be stored.
    *
-   * @return the {@link List} of {@link Metacard}
+   * @return the {@link Set} of {@link Metacard}
    */
-  public List<Metacard> getMetacards();
+   Set<Metacard> getMetacards();
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-api/src/main/java/ddf/catalog/operation/CreateRequest.java]: ❗️ Please note that this interface is whitelisted and cannot be changed without architectural input and a plan of action for deprecation/evolution of the contract.

(https://github.com/codice/ddf/blob/master/distribution/docs/src/main/resources/content/_whitelists/catalog-whitelist.adoc)

File: catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/CreateRequestImpl.java [hunk 3]
```diff
@@ -66,7 +66,10 @@ public CreateRequestImpl(List<Metacard> metacards, Map<String, Serializable> pro
   public CreateRequestImpl(
       List<Metacard> metacards, Map<String, Serializable> properties, Set<String> destinations) {
     super(properties);
-    this.metacards = metacards;
+    this.metacards = new HashSet<>();
+    for (Metacard metacard : metacards) {
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/CreateRequestImpl.java]: ✏️ This isn't the most efficient way of doing this. Consider using the `addAll` method instead.

File: catalog/security/catalog-security-plugin/src/test/java/ddf/catalog/security/plugin/SecurityPluginTest.java [hunk 4]
```diff
@@ -241,7 +241,7 @@ private Subject setupMockSubject() {
     private List<Metacard> metacards;
 
     @Override
-    public List<Metacard> getMetacards() {
```
Comment (coyotesqrl) [inline catalog/security/catalog-security-plugin/src/test/java/ddf/catalog/security/plugin/SecurityPluginTest.java]: ❓ Were there any tests for this? Should new ones be added?

File: catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/CreateRequestImpl.java [hunk 5]
```diff
@@ -83,7 +86,7 @@ public CreateRequestImpl(
    * @see ddf.catalog.operation.CreateRequest#getMetacards()
    */
   @Override
-  public List<Metacard> getMetacards() {
+  public Set<Metacard> getMetacards() {
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/CreateRequestImpl.java]: ❓ Were there any tests for this? Should new ones be added? I see that we have tests with their own implementations of the interface. Was updating those sufficient?

File: catalog/security/catalog-security-plugin/src/test/java/ddf/catalog/security/plugin/SecurityPluginTest.java [hunk 6]
```diff
@@ -241,7 +241,7 @@ private Subject setupMockSubject() {
...
+    public Set<Metacard> getMetacards() {
       if (metacards == null) {
         metacards = new ArrayList<>();
```
Comment (coyotesqrl) [inline catalog/security/catalog-security-plugin/src/test/java/ddf/catalog/security/plugin/SecurityPluginTest.java]: ❗️ This should be a set, not a list.

File: catalog/core/catalog-core-api/src/main/java/ddf/catalog/operation/CreateRequest.java (hunk 2 above)
Comment (coyotesqrl) [inline catalog/core/catalog-core-api/src/main/java/ddf/catalog/operation/CreateRequest.java]: ❓ Does documentation need to be updated for this change?

File: catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/CreateRequestImpl.java (hunk 3 above)
Comment (coyotesqrl) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/CreateRequestImpl.java]: ✏️ Consider as an alternative:
```java
    this.metacards = metacards.stream()
        .collect(Collectors.toSet());
```

**Comment by coyotesqrl** [comment]
Did you try turning it off and back on again? :trollface:

**Comment by pvargas** [comment]
🔥

**Pull request closed without merging**